src/Symfony/Component/Form/EnumFormTypeGuesser.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form;
+
+use Symfony\Component\Form\Extension\Core\Type\EnumType;
+use Symfony\Component\Form\Guess\Guess;
+use Symfony\Component\Form\Guess\TypeGuess;
+use Symfony\Component\Form\Guess\ValueGuess;
+
+final class EnumFormTypeGuesser implements FormTypeGuesserInterface
+{
+    /**
+     * @var array<string, array<string, \ReflectionNamedType|false>>
+     */
+    private array $cache = [];
+
+    public function guessType(string $class, string $property): ?TypeGuess
+    {
+        if (!($propertyType = $this->getPropertyType($class, $property))) {
+            return null;
+        }
+
+        return new TypeGuess(EnumType::class, ['class' => $propertyType->getName()], Guess::HIGH_CONFIDENCE);
+    }
+
+    public function guessRequired(string $class, string $property): ?ValueGuess
+    {
+        if (!($propertyType = $this->getPropertyType($class, $property))) {
+            return null;
+        }
+
+        return new ValueGuess(!$propertyType->allowsNull(), Guess::HIGH_CONFIDENCE);
+    }
+
+    public function guessMaxLength(string $class, string $property): ?ValueGuess
+    {
+        return null;
+    }
+
+    public function guessPattern(string $class, string $property): ?ValueGuess
+    {
+        return null;
+    }
+
+    private function getPropertyType(string $class, string $property): \ReflectionNamedType|false
+    {
+        if (isset($this->cache[$class][$property])) {
+            return $this->cache[$class][$property];
+        }
+
+        try {
+            $propertyReflection = new \ReflectionProperty($class, $property);
+        } catch (\ReflectionException) {
+            return $this->cache[$class][$property] = false;
+        }
+
+        $type = $propertyReflection->getType();
+        if (!$type instanceof \ReflectionNamedType || !enum_exists($type->getName())) {
+            $type = false;
+        }
+
+        return $this->cache[$class][$property] = $type;
```

Comment:
Would it be better in terms of memory usage to only store the enum name here (i.e. the result of `$type->getName()`) instead of a full reflection object?

Reply:
My main concern with the current approach is that I am unsure if it plays well will long running processes where the guesser is used while serving many requests.